### PR TITLE
add multipart + url-encoded content; cleanup old files

### PR DESCRIPTION
**Pull request by tanner0101**

### CODE REVIEW

File: Sources/Vapor/Multipart/MultipartParser.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
+import CMultipartParser
+
+/// Parses multipart-encoded `Data` into `MultipartPart`s. Multipart encoding is a widely-used format for encoding/// web-form data that includes rich content like files. It allows for arbitrary data to be encoded
```
Comment (MrLotU) [inline Sources/Vapor/Multipart/MultipartParser.swift]: I think you missed a newline here.
```suggestion
/// Parses multipart-encoded `Data` into `MultipartPart`s. Multipart encoding is a widely-used format for encoding
/// web-form data that includes rich content like files. It allows for arbitrary data to be encoded
```

**Comment by tanner0101** [comment]
@MrLotU the `CMultipartParser` lib is used to implement Vapor's new streaming multipart decoder. :)

**Comment by MrLotU** [comment]
@tanner0101 Oi, I got that much, but from my perspective it might be better in a separate package? Unless there is 0 reason to ever use it outside of the main package of course.

**Comment by tanner0101** [comment]
@MrLotU ah I see. Having everything in one package is a lot easier for development. But honestly, I'm not sure if there's much utility to separating it back out. Right now, vapor/multipart is separate, but that comes at the cost of maintenance burden and loss of whole-module optimizations. It's worth discussing more for sure.

**Comment by penny-coin** [comment]
Hey @tanner0101, you just merged a pull request, have a coin! 

You now have 1185 coins.

**Pull request merged**